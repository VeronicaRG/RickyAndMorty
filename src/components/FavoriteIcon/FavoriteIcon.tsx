import React from 'react';
import { FavoriteIconProps } from './FavoriteIcon.types';
import { Icon } from '@components/Icon';
import { TouchableOpacity } from 'react-native';

export const FavoriteIcon: React.FC<FavoriteIconProps> = ({
  isFavorite,
  widthIcon = 36,
  heightIcon = 36,
  handlePressFavorite,
  iconColor = '#ffff',
  testID = '',
}) => {
  return (
    <TouchableOpacity onPress={handlePressFavorite} testID={`${testID}favorite-icon-button`}>
      {isFavorite ? (
        <Icon
          testID={`${testID}favorite-icon-colorful`}
          width={widthIcon}
          height={heightIcon}
          name="fav-colorful"
        />
      ) : (
        <Icon
          color={iconColor}
          width={widthIcon}
          height={heightIcon}
          testID={`${testID}favorite-icon-monochrome`}
          name="fav-monochrome"
        />
      )}
    </TouchableOpacity>
  );
};
