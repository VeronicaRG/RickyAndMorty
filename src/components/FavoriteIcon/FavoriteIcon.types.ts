export interface FavoriteIconProps {
  iconColor?: string;
  widthIcon?: number;
  heightIcon?: number;
  isFavorite: boolean;
  handlePressFavorite: () => void;
  testID?: string;
}
