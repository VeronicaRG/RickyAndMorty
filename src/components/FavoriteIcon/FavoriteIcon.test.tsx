import React from 'react';
import { FavoriteIcon } from '@components/FavoriteIcon';
import { render, fireEvent } from '@testing-library/react-native';

describe('Component: FavoriteIcon', () => {
  it('should render and match snapshot with favorite true', () => {
    const { toJSON } = render(<FavoriteIcon isFavorite handlePressFavorite={() => {}} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot with favorite false', () => {
    const { toJSON } = render(<FavoriteIcon isFavorite={false} handlePressFavorite={() => {}} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the correct icon with colorful ID', () => {
    const { getByTestId } = render(<FavoriteIcon isFavorite handlePressFavorite={() => {}} />);

    expect(getByTestId('favorite-icon-colorful')).toBeTruthy();
  });

  it('should render the correct icon with monochrome ID', () => {
    const mockedFunction = jest.fn();
    const { getByTestId } = render(
      <FavoriteIcon isFavorite={false} handlePressFavorite={mockedFunction} />,
    );

    expect(getByTestId('favorite-icon-monochrome')).toBeTruthy();
  });

  it('should be render ...', () => {
    const mockedFunction = jest.fn();
    const { getByTestId } = render(
      <FavoriteIcon isFavorite handlePressFavorite={mockedFunction} />,
    );

    const isFavorite = getByTestId('favorite-icon-button');
    fireEvent.press(isFavorite);
    expect(mockedFunction).toHaveBeenCalled();
  });
});
