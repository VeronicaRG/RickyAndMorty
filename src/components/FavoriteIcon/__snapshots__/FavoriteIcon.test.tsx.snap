// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FavoriteIcon should render and match snapshot with favorite false 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="favorite-icon-button"
>
  <SvgMock
    color="#ffff"
    height={36}
    testID="favorite-icon-monochrome"
    width={36}
  />
</View>
`;

exports[`Component: FavoriteIcon should render and match snapshot with favorite true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="favorite-icon-button"
>
  <SvgMock
    color="#fff"
    height={36}
    testID="favorite-icon-colorful"
    width={36}
  />
</View>
`;
